### ISSUE TITLE
Проектирование экрана этапа Разработки

### ISSUE THREAD

**Issue by Smalex911** [GameDevTeamPerm/IT-manager#10]
В одном из визуальных редакторов спроектировать данный экран, на котором должен изображаться ход разработки, отображающий процесс начала и завершения разработки модуля. Также должно быть отображено количество прошедших дней с даты начала проекта, которое увеличивается со временем разработки модулей. Пользователь должен непосредственно видеть процесс разработки. Модули в упорядоченном пользователем списке постепенно с течением времени превращаются в разработанные. После того, как все модули разработаны, отображать пользователю информацию о затраченном времени на разработку и отображать кнопку перехода на следующий этап.

**Comment by TanakiSinitsu** [comment]
![module development](https://user-images.githubusercontent.com/18738823/31851315-b73e4bca-b67c-11e7-8d99-0d2d420df0b9.png)
PSD: https://drive.google.com/open?id=0B5YBwUye_JS9TENEb0YyYU15eFU

**Comment by TanakiSinitsu** [comment]
Удовлетворяет ли вас в общая стилистика приложения? Мне нужно знать, чтобы потом шаблон лишний раз не переделывать для следующих этапов.

**Comment by Smalex911** [comment]
Стилистика нормальная, но то, что изображено на скриншотах не отображает нужды этапа.
1. Нет временной шкалы, которая бы показывала пользователю, какой конкретно модуль разрабатывается в данный момент.
1. Также не отображена параллельная разработка, когда разные программисты разрабатывают разные модули.
1. Для каждого модуля должно быть видно, кто его разрабатывает (либо в виде столбца или строки разработчика, либо указывать в описании модуля).
1. Верхнюю статусную строку будет достаточно сложно переделать, имеет ли в этом оправданный смысл? Возможно дополнительные кнопки можно разместить где-то в другом месте.
1. В перечне этапов выбор проекта, найм сотрудников и результат не должны входить в состав этапов, они являются отдельными модальными окнами.
1. Колонка с этапами занимает 1/5 части экрана, она носит информативный характер, тем не менее занимает большую часть полезного места на экране.
1. Интерфейс должен быть похож на этап Планирования, только на планировании пользователь может перетаскивать модули и назначать сотрудникам, а на Разработке нет — у него на временной шкале должна появиться линия времени, которая будет продвигаться по мере прохождения разработки. При возникновении каких-либо событий время останавливается и выскакивает модальное окно с выбором ответа.
1. Интерфейс получился очень строгий, необходимо его чем-нибудь разбавить. Не вижу ни одной картинки.

**Comment by TanakiSinitsu** [comment]
1. В задании нет ни слова про создание временной шкалы.

> количество прошедших дней с даты начала проекта, которое увеличивается со временем разработки модулей

2. Не описано в задании. Нужно об этом догадаться?
3. Исправлено.
4. Исправлено.
5. Исправлено.
6. Подумаю, как поменять.
7. Жду этап Планирования, чтобы сделать этап Разработки похожим на него.
8. Это учебная программа, а не комикс.

*Линия времени - это диаграмма Ганта? Значит ли это, что нужно спроектировать экран с анимированной диаграмма Ганта, где пользователь будет нажимать всплывающие окна для подтверждения или отклонения статуса задач?

**Comment by Smalex911** [comment]
1. Можно и без временной шкалы, но главное чтобы было видно, что вот эти запланированные модули прошли разработку, а другие нет. Также по мере возникновения событий план модулей может меняться.
1. Вообще уже пора потихоньку вникать в суть процесса игры, и выполнять задания не только, опираясь на описание задачи, но и вообще логику игры. Несколько сотрудников параллельно могут выполнять разработку разных модулей. Модули могут различаться по продолжительности. Модуль может начаться в любое время независимо от других модулей разрабатываемых другими сотрудниками. Если есть вопросы, то лучше об этом сразу писать в слаке, а не тут обсуждать.
1. Сейчас мы собираем прототипы интерфейсов по каждому из этапов, чтобы потом уже сводить их все между собой к одному стилю, поэтому не нужно ждать кого-либо. Неизвестно сколько ты можешь прождать предыдущий этап, так как, чтобы добиться совершенства даже по одному из этапов нужно большое кол-во времени. К тому же вы сейчас живете в одной комнате и вам бы следовало делать работу вместе, помогая и советуясь друг с другом.
1. Мы делаем деловую игру, для пользователя она должна выглядеть как игра, и притягивать визуальным интерфейсом в первую очередь, если он посмотрит на скучные таблицы и списки с большими данными, он явно не захочет продолжать в нее играть. Первую оценку пользователь делает, основываясь на картинке, которую он видит, а уже потом на содержании.
1. Пока, учитывая критерии связанные с временной линией, я представляю диаграмму Ганта, но это не окончательный вариант, если есть какие-то другие варианты, которые будут учитывать большинство критериев, то я готов рассмотреть их.

**Comment by TanakiSinitsu** [comment]
![module development](https://user-images.githubusercontent.com/18738823/32049037-eeee5bf0-ba64-11e7-9054-57869fcf346a.png)

PSD: https://drive.google.com/open?id=0B5YBwUye_JS9QjdxTExQYkJiX3c

**Comment by Smalex911** [comment]
В целом хорошая работа. Есть пару моментов на доработку.
1. Есть ли смысл выделять под каждый модуль кнопку Разработать, если разработка возможна только по времени согласно установленному плану.
1. Может попробовать в диаграмме указывать слева в строках сотрудников, которые задействованы в разработке, а не задачи. Иначе не видно, кто из сотрудников простаивает, пока другие работают.
1. Еще я не понял по какому принципу происходит разработка в прошлом (на скриншотах красный модуль)
1. Спустя пару минут я догадался куда нужно смотреть, чтобы определить какой модуль за каким идет. Нужно будет сделать подсказку пользователю, но об этом потом. Я потом посмотрю, если в диаграмме Ганта можно будет рисовать зависимости, то верхнюю часть с модулями можно будет сократить.

Принцип этапа разработки в том, что пользователь по составленному на этапе планирования плану (в виде диаграммы Ганта или что-то иное, главное чтобы порядок был виден) следит за ходом разработки. Другими словами проходит симуляция разработки согласно плану. Сейчас этот этап проходит безболезненно для пользователя, внутри игры только генерятся ошибки, а в будущем будут возникать события, которые будут влиять на установленный план. Скорее всего, если возникнут такие события, то пользователя нужно будет перекинуть снова на планирование, где он сможет передвинуть блоки и продолжить разработку.

**Comment by TanakiSinitsu** [comment]
Красный модуль идёт сначала временной шкалы, так как он не связан с другими модулями и идёт параллельно им. То есть мы могли его начать разрабатывать первым и тогда был вверху, а остальные под ним. Такое может происходить по плану игры?

**Comment by Smalex911** [comment]
Да, может. Только не уверен, что пользователю будет понятно каким образом он возвращается в прошлое. Нужно провести эксперимент на ком-нибудь
